app/script/event/EventRepository.coffee

```diff
@@ -190,20 +187,34 @@ class z.event.EventRepository
         else
           @logger.log @logger.levels.ERROR, "Failed to get notifications: #{error_response.message}", error_response
           reject new z.event.EventError z.event.EventError::TYPE.REQUEST_FAILURE
-
   ###
   Get the last notification ID for a given client.
   @note This API endpoint is currently broken on the backend
   @return [Promise] Promise that resolves with the last known notification ID matching a client
   ###
   get_last_notification_id: ->
-    return new Promise (resolve, reject) =>
-      @notification_service.get_notifications_last @current_client?().id
-      .then (response) ->
-        resolve response.id
-      .catch reject
+    @notification_service.get_notifications_last @current_client?().id
+    .then (response) ->
+      return response.id
+
   ###
-  Will retrieve missed notifications from the stream after a connectivity loss.
+  Get the last notification.
+  @return [Promise] Promise that resolves with the last handled notification ID
+  ###
+  get_last_notification_id_from_db: (last_notification_id) =>
+    @notification_service.get_last_notification_id_from_db()
+    .then (last_notification_id) =>
+      @last_notification_id last_notification_id
+
+      @last_notification_id.subscribe (last_notification_id) =>
+        @logger.log @logger.levels.INFO, "Last notification ID updated to '#{last_notification_id}'"
+        @notification_service.save_last_notification_id_to_db last_notification_id if last_notification_id
+
```

Comment:
I think there is an issue here with the subscription. Whenever `get_last_notification_id_from_db` is executed it will add another subscription. Means when you executed `get_last_notification_id_from_db ` a second time and change `last_notification_id`, the handler will fire twice.